Backend/src/main/java/com/abhinternship/auctionapp/service/ProductService.java

```diff
@@ -116,4 +115,34 @@ public Product update(Integer id, Product request) {
     public List<Double> getAllPrices() {
         return repository.getAllByOrderByHighestBid();
     }
-}
+
+    public List<Product> getAllActiveProductsByUser(User user, Long pageSize) throws RepositoryException {
+        Pageable pageable = PageRequest.of(0, Math.toIntExact(pageSize));
```

Comment:
Implement pagination, wherever you are using pageable without incrementing the page number.